app/src/main/java/edu/wpi/first/shuffleboard/app/components/DashboardTabPane.java

```diff
@@ -224,8 +224,8 @@ private boolean shouldAutopopulate(String sourceId) {
      */
     private boolean noExistingWidgetsForSource(String id) {
       for (WidgetTile tile : getWidgetPane().getTiles()) {
-        DataSource<?> source = tile.getWidget().getSource();
-        if (id.startsWith(source.getId())) {
+        boolean match = tile.getWidget().getSources().stream().anyMatch(s -> id.startsWith(s.getId()));
```

Comment:
this whole function can be

`getTiles.stream().allMatch(tile -> !tile.getWidget().getSource.stream().anyMatch(s -> id.startsWith(s.getId()))`